show-coordinates-in-multiple-formats/src/main/java/com/esri/arcgismaps/sample/showcoordinatesinmultipleformats/screens/CoordinatesLayout.kt

```diff
@@ -0,0 +1,115 @@
+import androidx.compose.foundation.layout.ime
+import androidx.compose.foundation.text.KeyboardActions
+import androidx.compose.foundation.text.KeyboardOptions
+import androidx.compose.material3.OutlinedTextField
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.State
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.rememberUpdatedState
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.focus.FocusRequester
+import androidx.compose.ui.focus.focusRequester
+import androidx.compose.ui.platform.LocalDensity
+import androidx.compose.ui.platform.LocalFocusManager
+import androidx.compose.ui.text.input.ImeAction
+import androidx.compose.ui.text.input.KeyboardType
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.lifecycle.viewmodel.compose.viewModel
+import com.esri.arcgismaps.sample.showcoordinatesinmultipleformats.components.MapViewModel
+
+@Composable
+fun CoordinatesLayout(mapViewModel: MapViewModel) {
+    CoordinateField(
+        notationType = MapViewModel.NotationType.DD,
+        text = mapViewModel.decimalDegrees,
+        labelText = "Decimal Degrees",
+        onTextChanged = mapViewModel::setDecimalDegreesCoordinate,
+        onQuerySubmit = mapViewModel::fromCoordinateNotationToPoint,
+    )
+
+    CoordinateField(
+        notationType = MapViewModel.NotationType.DMS,
+        text = mapViewModel.degreesMinutesSeconds,
+        labelText = "Degrees, Minutes, Seconds",
+        onTextChanged = mapViewModel::degreesMinutesSecondsCoordinate,
+        onQuerySubmit = mapViewModel::fromCoordinateNotationToPoint,
+    )
+
+    CoordinateField(
+        notationType = MapViewModel.NotationType.UTM,
+        text = mapViewModel.utm,
+        labelText = "UTM",
+        onTextChanged = mapViewModel::setUTMCoordinate,
+        onQuerySubmit = mapViewModel::fromCoordinateNotationToPoint,
+    )
+
+    CoordinateField(
+        notationType = MapViewModel.NotationType.USNG,
+        text = mapViewModel.usng,
+        labelText = "USNG",
+        onTextChanged = mapViewModel::setUSNGDegreesCoordinate,
+        onQuerySubmit = mapViewModel::fromCoordinateNotationToPoint,
+    )
+}
+
+@Composable
+fun CoordinateField(
+    modifier: Modifier = Modifier,
+    notationType: MapViewModel.NotationType,
+    text: String = "",
+    onTextChanged: (String) -> Unit,
+    onQuerySubmit: (MapViewModel.NotationType, String) -> Unit,
+    labelText: String
+) {
+    // remember the OutlinedTextField's focus requester to change focus on search
+    val focusRequester = remember { FocusRequester() }
+    // focus manager is used to clear focus from OutlinedTextField on search
+    val focusManager = LocalFocusManager.current
+    // set the keyboard options for all field to search
+    val keyboardOptions = KeyboardOptions(
+        keyboardType = KeyboardType.Text,
+        imeAction = ImeAction.Search
+    )
+
+    OutlinedTextField(
```

Comment:
This is a nitpick, but on emulator I keep trying to press enter to search when changing the coordinates, and it puts in a new line, but I think you could set `singleLine = true` here and at least it would not add the new line :D

Reply:
`singleLine = true` will display the text field in a single line, but while testing I noticed that on pressing enter the text will still be added into a new line (but just not visible) and will not submit query. Instead I added a `onKeyEvent` property to the modifier to detect an enter clicked to submit the query notation and coordinate.